sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -284,8 +284,12 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    * @since 2.0.0
    */
   override def dropTempView(viewName: String): Unit = {
-    sparkSession.sharedState.cacheManager.uncacheQuery(sparkSession.table(viewName))
-    sessionCatalog.dropTable(TableIdentifier(viewName), ignoreIfNotExists = true, purge = false)
+    val maybeTempView = sparkSession.sessionState.catalog.getTempView(viewName)
+    if (maybeTempView.isDefined) {
+      val view = SubqueryAlias(viewName, maybeTempView.get, Some(TableIdentifier(viewName)))
```

Comment:
do you mean you pulled my PR and remove the subquery and this test still passed?